confgenerator/logging_processors.go

```diff
@@ -39,77 +44,184 @@ func init() {
 	LoggingProcessorTypes.RegisterType(func() Component { return &LoggingProcessorParseJson{} })
 }
 
-// A LoggingProcessorParseRegex applies a regex to the specified field, storing the named capture groups as keys in the log record.
-type LoggingProcessorParseRegex struct {
-	ConfigComponent             `yaml:",inline"`
-	LoggingProcessorParseShared `yaml:",inline"`
+// A LoggingProcessorParseRegexSimple applies a regex to the specified field, storing the named capture groups as keys in the log record.
+// This was maintained in addition to the parse_regex_complex to ensure backward compatibility with any existing configurations
+type LoggingProcessorParseRegexSimple struct {
+	ConfigComponent        `yaml:",inline"`
+	fluentbit.ParserShared `yaml:",inline"`
+	Field                  string `yaml:"field,omitempty"`
 
 	Regex string `yaml:"regex,omitempty" validate:"required"`
 }
 
-func (r LoggingProcessorParseRegex) Type() string {
+func (r LoggingProcessorParseRegexSimple) Type() string {
```

Comment:
This "simple" regex parser was left in place as I wanted this update to not break previously configured one-off regex parsers. If that's not really a concern it could be removed leaving only the multiple parser version.

Reply:
Resolved.